QuizApp/LeaderboardViewController.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  LeaderboardViewController.swift
+//  QuizApp
+//
+//  Created by five on 16.05.2021..
+//
+
+import Foundation
+import UIKit
+
+class LeaderboardViewController : UIViewController {
+    let cellIdentifier = "leaderborderCell"
+    let router: AppRouterProtocol
+    let networkService: NetworkServiceProtocol = NetworkService()
+    
+    let quizId: Int
+    var results = [LeaderboardResult]()
+    
+    var boardLabel: UILabel!
+    var closeButton: UIButton!
+    var table: UITableView!
+    
+    init(router: AppRouterProtocol, quizId: Int) {
+        self.router  = router
+        self.quizId = quizId
+        
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        buildViews()
+    }
+    
+    func buildViews() {
+        view.backgroundColor = .systemBlue
+        
+        boardLabel = UILabel()
+        boardLabel.text = "Leaderboard"
+        boardLabel.textColor = .white
+        boardLabel.font = boardLabel.font.withSize(30)
+        view.addSubview(boardLabel)
+        
+        boardLabel.snp.makeConstraints {
+            $0.top.equalToSuperview().offset(50)
+            $0.centerX.equalToSuperview()
+        }
+        
+        closeButton = UIButton()
+        closeButton.setImage(UIImage(systemName: "xmark"), for: .normal)
+        closeButton.tintColor = .white
+        closeButton.addTarget(self, action: #selector(hideLeaderborder), for: .touchUpInside)
+        view.addSubview(closeButton)
+        
+        closeButton.snp.makeConstraints {
+            $0.top.equalTo(boardLabel.snp.top)
+            $0.trailing.equalToSuperview().inset(20)
+        }
+        
+        table = UITableView()
+        view.addSubview(table)
+        table.allowsSelection = false
+        table.isScrollEnabled = false
+        table.separatorColor = .white
+        table.backgroundColor = .systemBlue
+        
+        table.register(UITableViewCell.self, forCellReuseIdentifier: cellIdentifier)
+        
+        let headerView = UIView()
+        table.tableHeaderView = headerView
+
+        headerView.backgroundColor = .systemBlue
+        headerView.snp.makeConstraints {
+            $0.height.equalTo(50)
+            $0.width.equalToSuperview()
+            $0.leading.equalToSuperview()
+        }
+        
+        
+        table.dataSource = self
+        table.snp.makeConstraints {
+            $0.top.equalTo(boardLabel.snp.bottom).offset(10)
+            $0.leading.equalToSuperview()
+            $0.width.equalToSuperview()
+            $0.bottom.equalToSuperview()
+        }
+        
+        networkService.fetchLeaderboard(quizId: quizId) { results, error in
```

Comment:
This is the first place I've noticed this, but please check the whole code.
`fetchLeaderboard` has an escaping closure for its parameter. That means that that closure will outlive the function and create a memory leak if you're using `strong self` here (which you are on the line 93).
What will  happen is following:
closure that's referencing `self` will increase its reference count and if that closure doesn't get deallocated  before the `LeaderboardViewController` it will keep it in the memory while you think  that it's gone.

This is explained in the materials you got  for Memory Management in Swift.
To remedy this you can do the following:
```swift
networkService
    .fetchLeaderboard(quizId: quizId) { [weak self] results, error in 
        guard let self = self else { return }
    
        // the rest of the code
    }
```